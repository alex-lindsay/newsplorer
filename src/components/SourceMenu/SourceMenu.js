import React from "react";
import styles from "./SourceMenu.module.css";

import IconButton from "@material-ui/core/IconButton";
import BusinessIcon from "@material-ui/icons/Business";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import Tooltip from "@material-ui/core/Tooltip";

const SourceMenu = props => {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const handleMenu = event => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = event => {
    setAnchorEl(null);
  };

  const menuItems = props.sources
    ? props.sources
        .sort((l, r) => (l.name < r.name ? -1 : 1))
        .map(source => (
          <Tooltip
            key={source.name}
            title={source.description}
            placement="right"
            aria-label={source.description}
          >
            <MenuItem onClick={handleClose} data-source={source.name}>
              {source.name}
            </MenuItem>
          </Tooltip>
        ))
    : null;
  return (
    <div className={styles.SourceMenu} data-testid="source-menu">
      <IconButton
        aria-label="account of current user"
        aria-controls="menu-sourcemenu"
        aria-haspopup="true"
        onClick={handleMenu}
        color="inherit"
      >
        <BusinessIcon />
      </IconButton>
      <Menu
        id="menu-sourcemenu"
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: "top",
          horizontal: "right",
        }}
        keepMounted
        transformOrigin={{
          vertical: "top",
          horizontal: "right",
        }}
        open={open}
        onClose={handleClose}
      >
        <MenuItem key="-" onClick={handleClose} data-source="">
          None
        </MenuItem>
        {menuItems}
      </Menu>
    </div>
  );
};

export default SourceMenu;
