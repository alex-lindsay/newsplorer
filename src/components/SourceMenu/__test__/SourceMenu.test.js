import React from "react";
import { render, cleanup } from "@testing-library/react";
import SourceMenu from "../SourceMenu";

afterEach(cleanup);

it("renders App without crashing", () => {
  const { getByTestId } = render(<SourceMenu />);
  expect(getByTestId("source-menu")).toBeInTheDocument();
});

it("matches expected snapshot", () => {
  const { asFragment } = render(<SourceMenu />);
  expect(asFragment()).toMatchSnapshot();
});
