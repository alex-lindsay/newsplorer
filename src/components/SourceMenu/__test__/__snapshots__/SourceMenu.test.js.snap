// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected snapshot 1`] = `
<DocumentFragment>
  <div
    class="SourceMenu"
    data-testid="source-menu"
  >
    <button
      aria-controls="menu-sourcemenu"
      aria-haspopup="true"
      aria-label="account of current user"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
